File: webapp/littleId/littleId/src/main/scala/littleware/apps/littleId/client/test/PackageTestSuite.scala
/*
 * Copyright 2011 Reuben Pasquini All rights reserved.
 * 
 * The contents of this file are available subject to the terms of the
 * Lesser GNU General Public License (LGPL) Version 2.1.
 * http://www.gnu.org/licenses/lgpl-2.1.html.
 */


package littleware.apps.littleId.client.test

import com.google.inject.Inject
import java.util.logging.Level
import junit.framework.TestSuite
import littleware.base.AssertionFailedException
import littleware.bootstrap.client.AppBootstrap
import littleware.scala.LazyLogger
import littleware.test.TestFactory


class PackageTestSuite @Inject()(
  verifyTester:VerifyToolTester,
  jaasTester:JaasLoginTester
) extends TestSuite {
  setName( getClass.getName )
  addTest( verifyTester )
  addTest( jaasTester )
}

object PackageTestSuite {
  val log = LazyLogger( getClass )
  def suite():TestSuite = try {
    (new TestFactory()).build(
      AppBootstrap.appProvider.get().build(),
      classOf[PackageTestSuite]
    )
  } catch {
    case ex => {
      log.log( Level.WARNING, "Failed to launch test suite", ex )
      throw new AssertionFailedException( "Failed to launch test", ex )
    }
  }
}

File: littleware/src/littleware/lgo/test/PackageTestSuite.java
/*
 * Copyright 2010 Reuben Pasquini All rights reserved.
 * 
 * The contents of this file are subject to the terms of the
 * Lesser GNU General Public License (LGPL) Version 2.1.
 * http://www.gnu.org/licenses/lgpl-2.1.html.
 */

package littleware.lgo.test;

import com.google.inject.Inject;
import com.google.inject.Provider;
import junit.framework.Test;
import junit.framework.TestSuite;
import littleware.bootstrap.AppBootstrap;
import littleware.test.TestFactory;

/**
 * littleware.apps.lgo package test suite
 */
public class PackageTestSuite extends TestSuite {

    @Inject
    public PackageTestSuite( Provider<ArgParserTester> factoryArgTester,
            Provider<XmlLgoHelpTester> factoryXmlHelpTester
            )
    {
        setName( PackageTestSuite.class.getName() );

        if ( true ) {
            this.addTest( factoryArgTester.get() );
            this.addTest( factoryXmlHelpTester.get() );
        }
    }

    /**
     * Return the singleton set by the constructor as part
     * of the OSGi bootstrap process
     */
    public static Test suite() {
        return (new TestFactory()).build(
                AppBootstrap.appProvider.get().build(),
                PackageTestSuite.class
                );
    }

    public static void main( String[] args ) {
        junit.swingui.TestRunner.main( new String[] {
            "-noloading",
            PackageTestSuite.class.getName()
            }
        );
    }

}
